cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/StageSubmitter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api;
+
+import co.cask.cdap.api.messaging.MessagingContext;
+
+/**
+ * Similar to {@link TransformContext}, but also exposing functionality of {@link MessagingContext}.
+ *
+ * @param <T> execution context
+ */
+public interface StageSubmitter<T> extends MessagingContext {
```

Comment:
This seems like an unnecessary interface. Shouldn't it be another context? Everywhere else, plugins are just given a context object.